src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SslOverTdsStream.NetStandard.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.SqlClient.SNI
+{
+    internal sealed partial class SslOverTdsStream : Stream
+    {
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            if (_encapsulate)
```

Comment:
Encapsulation is only going to be enabled for a single packet for the lifetime of a connection. This is another case where I'd be quite happy to do silly things to get the absolute fastest check and call for the common case, which is passthrough. Since I hadn't measured that particular perf I wrote it the same way as the original. Do you think there's any particular perf merit one way or the other? Ideally this is a case where it'd be nice to have something like a tail call happen if we end up just passing through to the `_stream` implementation but that'd be a JIT thing.

Reply:
Oh no, I wasn't referring to any perf-related point here - only a code styling suggestion. Basically you have the main method body in an if, and then a tiny else that just does return. In these I cases I tend to invert the condition, to get "rid" of the else block early.

So instead of:

```c#
if (_encapsulate) {
    // Long block
} else {
    return await _stream.ReadAsync(buffer, offset, count, cancellationToken);
}
```

I tend to write:

```c#
if (!_encapsulate)
    return await _stream.ReadAsync(buffer, offset, count, cancellationToken);

// Long block
```

This makes code a bit easier to follow and reduces an indentation level for the long block. Just a style suggestion.